### ISSUE TITLE
JBullet gradle.build causing NetBeans 11 Gradle error

### ISSUE THREAD

**Issue by louhy** [jMonkeyEngine/jmonkeyengine#1105]
I get this error due to a relative path: srcDir '../jme3-bullet/src/common/java' 

![Screenshot from 2019-06-02 02-04-37](https://user-images.githubusercontent.com/41243237/58757455-d2730100-84da-11e9-94bd-c679b9c371bd.png)

Removing that makes this error go away. I suppose that's a bug in NB Gradle, but I wonder why it's not just a project dependency? 

```groovy
sourceSets {
    main {
        java {
            srcDir 'src/main/java'
            srcDir '../jme3-bullet/src/common/java' <--compile project('jme3-bullet') ?
        }
    }
}

dependencies {
    compile 'java3d:vecmath:1.3.1'
    compile 'jbullet:jbullet'
    compile 'stack-alloc:stack-alloc'  <--doesn't exist? net.sf.sociaal is actual group
    compile project(':jme3-core')
    compile project(':jme3-terrain')
}
```

2nd issue area of concern...
With that "stack-alloc" dependency, whatever it is, it doesn't even appear to be used. 
I noticed it when I was getting this error out of Gradle: 
>  Could not resolve all files for configuration ':runtimeClasspath'.
>     Could not find jbullet:jbullet:.
>      Required by:
>          project : > org.jmonkeyengine:jme3-jbullet:3.3.0-alpha2
>    **> Could not find stack-alloc:stack-alloc:.**
>      Required by:
>          project : > org.jmonkeyengine:jme3-jbullet:3.3.0-alpha2

Seems safe to remove the "dependency". 

I don't see stack-alloc.jar itself (inside jmonkeyengine/lib) referenced anywhere else, either. Looking at Git I see it _used_ to be directly referenced from the /lib folder before, but now it's that weird project dependency. 

I'll submit a PR to clean these up if others are in agreement.

**Comment by quazi-irfan** [comment]
jBullet uses stack-alloc. [Here](http://www.java-gaming.org/topics/jstackalloc-stack-allocation-of-quot-value-quot-objects-in-java/18843/view.html) I am curious - why you are adding jBullet source in your project?

**Comment by louhy** [comment]
Might have a partial reason for the relative path situation. There seems to be no info in Git, GitHub, or the forum on this, but jme3-bullet was originally added as a project dependency for a few days and then reverted back to use the relative path. 

When you're using jbullet, having jme3-bullet as a Gradle project dependency appears to be triggering jme3-desktop to load bullet natives. I guess maybe that's the issue. You can include bullet-native as a project dependency along with jbullet to stop it from _crashing_ (but that's silly). 

So that might need to stay unless we can remove the jbullet dependency on bullet. I feel like dependency injection should be involved here instead of having that dependency, but I still have a ways to go to understand how this is wired up.

**Comment by pspeed42** [comment]
There are common sources that are used by both jme bullet-native and jme-jbullet... and I guess they are interdependent so you can't compile the common classes on their own or something.  There is a reason and it's something along those lines.

**Comment by louhy** [comment]
I'll look over the link later. I haven't seen anything bad from commenting out and recompiling jbullet so far, and I can't find a single reference to it. Maybe it used to, but no longer does? (At least the version I have as a dependency.) 

> why you are adding jBullet source in your project?

I'm not, that's the jBullet build.gradle shown.

**Comment by louhy** [comment]
@pspeed42 Yeah something like that is going on. I'm going to work on gaining some better understanding.

**Comment by stephengold** [comment]
The jbullet code uses stack-alloc.

**Comment by louhy** [comment]
Just seeing that now. jme3-jbullet doesn't use it, jbullet itself does. So that makes it a transitive dependency which shouldn't be needed in the jbullet gradle.build.

**Comment by louhy** [comment]
Well there's now an issue filed for NetBeans to fix the "node" error for that srcDir path. 

On #2, I do still think that stack-alloc dependency isn't needed, and I'm uncertain how it's even being located. I don't see a "stack-alloc" group here: https://bintray.com/search?query=stack-alloc or even a jbullet:jbullet groupId:artifactId. 

There are so many "jbullet" binaries out there, not to mention GitHub repos, it's hard to figure out which one JME is actually supposed to be using--or which source. Tracking down the history of this thing seems like solving a mystery. 

How are you guys keeping track and making sense of it all? Am I looking in the wrong place?

**Comment by stephengold** [comment]
I believe the JME3 build scripts use the `jbullet.jar` and `stack-alloc.jar` committed to the JME3 repo:
    https://github.com/jMonkeyEngine/jmonkeyengine/tree/master/lib

I believe `jbullet.jar` was built from this codebase:
    https://github.com/normen/jbullet

In order to re-build it using JDK8, I had to use an updated `stack-alloc.jar`, so I created my own fork:
    https://github.com/stephengold/jbullet

**Comment by louhy** [comment]
Ahh, that helps a lot. Still kinda sucks that I couldn't compile with the NB 11 Gradle though - that lib gave me the error I pasted above. I'll continue exploring until I "get" it.

As you can probably tell, I'm just a displaced Ant guy still adjusting to Gradleese.

**Comment by louhy** [comment]
I'm coming to the conclusion that NB11's integrated Gradle support is just incapable of supporting JME builds (as of today). In addition to breaking NB's project view pictured here, Apparently it's also causing some classloader chaos (which would explain some of the localized weirdness I've been seeing). 11 seems to be fine for _projects which use JME_, but for building JME itself (with NetBeans) **we're better off sticking to NB10 or older and using the old Gradle plugin**. 

Eliminating the need to use relative src paths in the Gradle build would probably resolve this stuff, but that looks like it'd require some project refactoring/restructuring. 

So basically I've gone back to NB 10 for anything JME-lib related, and I'll only use 11 for JME projects. 

If we don't care about NB11 builtin Gradle support for JME builds (or will just hope the situation changes), issue can be closed. If eliminating these relative paths in JME's build is considered a good idea (and I think there's more than just JBullet's build relying on it...), that's probably the "resolution" for the issue. I do think eliminating the ../ paths would be best, but kind of a big job for little gain atm.  :grimacing:
